ci/package.sh

```diff
@@ -106,6 +106,101 @@ case "$OS" in
             "packages/$TARGET_DIR/"
         ;;
 
+    'mariner')
+        apt-get update
+        apt-get install -y git sudo software-properties-common
+        sudo add-apt-repository ppa:longsleep/golang-backports
+        apt-get update
+        apt -y install make tar wget curl rpm qemu-utils golang-1.13-go genisoimage pigz cpio python-pip python3-distutils
+        sudo ln -vs /usr/lib/go-1.13/bin/go /usr/bin/go
+        mv /.dockerenv /.dockerenv.old 
+
+        mkdir -p packages
+
+        git clone https://github.com/microsoft/CBL-Mariner.git
+        pushd CBL-Mariner
+        git checkout "1.0-stable"
+        pushd toolkit
+        make package-toolkit REBUILD_TOOLS=y
+        popd
+        mv out/toolkit-*.tar.gz "/src/CBL-Mariner/toolkit.tar.gz"
+        popd
+
+        make ARCH="$ARCH" PACKAGE_VERSION="$PACKAGE_VERSION" V=1 dist
+
+        MarinerRPMBUILDDIR=/src/Mariner-Build
+        MarinerSpecsDir=$MarinerRPMBUILDDIR/SPECS/aziot-identity-service
+        MarinerSourceDir=$MarinerSpecsDir/SOURCES
+
+        # Extract built toolkit in building direectory
+        mkdir -p $MarinerRPMBUILDDIR
+        pushd $MarinerRPMBUILDDIR
+        mv /src/CBL-Mariner/toolkit.tar.gz toolkit.tar.gz
+        tar -xzvf toolkit.tar.gz
+        popd
+
+        # move tarballed iot-identity-service source to building directory
+        mkdir -p $MarinerSourceDir
+        mv /tmp/aziot-identity-service-$PACKAGE_VERSION.tar.gz $MarinerSourceDir/aziot-identity-service-$PACKAGE_VERSION.tar.gz
+
+        mkdir /temp
+        rustup install stable
+        rustup default stable
+        wget https://github.com/eqrion/cbindgen/archive/refs/tags/v0.18.0.tar.gz -O /temp/cbindgen-0.18.0.tar.gz
```

Comment:
`install-build-deps.sh` uses `^0.15`, so it'll install the latest 0.15, not 0.18. Same for bindgen below, ^0.54 vs 0.57

It's fine to update to the latest deps but then you need to update `install-build-deps.sh` too.

Reply:
all references to 0.18 should not be gone and replace with 0.15

Reply:
You haven't changed `install-build-deps.sh`

Also you haven't changed bindgen.

Reply:
I might be wrong here but i dont think the bingen requirements from intsall-deps.sh will effect us here. Part of the mariner toolkit is that it works fully offline when building RPMS so it will only have access to the bindgen and cbindgen packages that we provided in the form of tarballs before kicking off the RPM build process. These are being downloaded in packages.sh. and then extracted by the spec file in the prep stage.

Reply:
The point is to use the same versions of dependencies regardless of whether it's a Mariner build or not. It has nothing to do with Mariner build being hermetic or not.

In fact, for the same reason, the `rustup install stable` and `rustup default stable` commands above this need to use the version defined in the `rust-toolchain` file.

Reply:
Ok i think i understand. so would we just need to have condition that checks if the build is for mariner and then instead of installing "^0.15" it would get exactly "0.15.0". and then the same thing for bingen.

Reply:
No, because that would still install different versions of dependencies, eg 0.15.0 on Mariner and 0.15.2 on non-Mariner.

You'd need to install 0.15.0 always. ie change `^0.15` to `=0.15.0` for everything.

Reply:
should be resolved now only using exact versions